hs/src/Reach/Connector/ALGO.hs

```diff
@@ -1300,8 +1329,18 @@ checkTxnInit vTypeEnum ct_mcsend = do
   -- [ txn ]
   return ()
 
+checkTxnUsage :: CheckTxn -> App ()
+checkTxnUsage (CheckTxn {..}) = do
+  case ct_mtok of
+    Just tok -> do
+      newToks <- (liftIO . readIORef) =<< asks eNewToks
+      when (tok `S.member` newToks) $ do
+            bad $ LT.pack $ runIdentity . flip runReaderT 3 $ getErrorMessage [] ct_at True Err_TransferNewToken
```

Comment:
That's a mouthful :)

Reply:
I think you can just do `nest` and `show` or something?

Reply:
I believe the indentation will be incorrect if using the `nest`. LMK if you think it's still worth converting to though

Reply:
Honestly, I'd rather have it look bad then expose these details of the Texty monad. `nest` twice would give 4 lines right? That's not so bad?